auth/app/Http/Controllers/AuthController.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Cache;
+use Laravel\Lumen\Routing\Controller as BaseController;
+use Lcobucci\JWT\Builder;
+use Lcobucci\JWT\Parser;
+use Lcobucci\JWT\Signer\Hmac\Sha256;
+use Miner\Auth\Entities\User;
+use Miner\Auth\Repositories\UserRepository;
+use Miner\Auth\Entities\Uuid;
+
+class AuthController extends BaseController
+{
+    public function login(Request $request, UserRepository $userRepository, Cache $cache)
+    {
+        $this->validate($request, [
+            'username' => 'required',
+            'password' => 'required'
+        ]);
+
+        $email = $request->get('email');
+        $password = $request->get('password');
+
+        $user = $userRepository->findByEmail($email);
+
+        // use php 5.5 password_hash/password_verify magix to validate the password matches
+        if(!password_verify($password, $user->getPasswordHash())) {
+            throw new \Exception('invalid password');
```

Comment:
We should have a discussion around when to create custom Exceptions. I am always torn on a base exception or explicit classes for each exception category (AuthPassword, AuthEmail, etc.)